system-tests/lib/cre/features/consensus/v1/consensus.go

```diff
@@ -0,0 +1,234 @@
+package v1
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/google/uuid"
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog"
+
+	chainselectors "github.com/smartcontractkit/chain-selectors"
+
+	capabilitiespb "github.com/smartcontractkit/chainlink-common/pkg/capabilities/pb"
+	kcr "github.com/smartcontractkit/chainlink-evm/gethwrappers/keystone/generated/capabilities_registry_1_1_0"
+	jobv1 "github.com/smartcontractkit/chainlink-protos/job-distributor/v1/job"
+
+	"github.com/smartcontractkit/chainlink-deployments-framework/offchain/jd"
+	"github.com/smartcontractkit/chainlink-deployments-framework/operations"
+	keystone_changeset "github.com/smartcontractkit/chainlink/deployment/keystone/changeset"
+	ks_contracts_op "github.com/smartcontractkit/chainlink/deployment/keystone/changeset/operations/contracts"
+	"github.com/smartcontractkit/chainlink/v2/core/services/gateway/config"
+
+	"github.com/smartcontractkit/chainlink/system-tests/lib/cre"
+	"github.com/smartcontractkit/chainlink/system-tests/lib/cre/contracts"
+	"github.com/smartcontractkit/chainlink/system-tests/lib/cre/don/jobs"
+	"github.com/smartcontractkit/chainlink/system-tests/lib/cre/don/jobs/ocr"
+	"github.com/smartcontractkit/chainlink/system-tests/lib/cre/features/consensus"
+)
+
+const flag = cre.ConsensusCapability
+
+type Consensus struct{}
+
+func (c *Consensus) Flag() cre.CapabilityFlag {
+	return flag
+}
+
+func (c *Consensus) PreEnvStartup(
+	ctx context.Context,
+	testLogger zerolog.Logger,
+	registryChainSelector uint64,
+	topology *cre.Topology,
+	creEnv *cre.Environment,
+	gatewayJobConfigs map[cre.NodeUUID]*config.GatewayConfig,
+) (*cre.PreEnvStartupOutput, error) {
+	capabilities := make(map[uint64][]keystone_changeset.DONCapabilityWithConfig)
+	for _, donMetadata := range topology.DonsMetadataWithFlag(flag) {
```

Comment:
This is in the wrong place -- adding a capability to the registry should be the last thing that happens, after it has been configured